core/src/main/java/cz/o2/proxima/storage/ThroughputLimiter.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.storage;
+
+import cz.o2.proxima.annotations.Evolving;
+import cz.o2.proxima.annotations.Internal;
+import java.io.Serializable;
+import java.time.Duration;
+import java.util.Collection;
+
+/** A limiter of data rate coming from various sources. */
+@Evolving
+public interface ThroughputLimiter extends Serializable {
+
+  @Internal
+  class NoOpThroughputLimiter implements ThroughputLimiter {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final ThroughputLimiter INSTANCE = new NoOpThroughputLimiter();
+
+    @Override
+    public Duration getPauseTime(Context context) {
+      return Duration.ZERO;
+    }
+  }
+
+  /** A context of {@link #getPauseTime} method. */
+  interface Context {
+
+    /**
+     * Get collection of {@link Partition Partitions} associated with reaeder reading with this
+     * limiter
+     *
+     * @return list of associated partitions
+     */
+    Collection<Partition> getConsumedPartitions();
+
+    /** Get count of all partitions in the reader. */
+    int getNumPartitions();
```

Comment:
is internal implementation doing anything else than `getConsumedPartitions().size()`? do we need this method?